spec/registry.md

```diff
@@ -1,11 +1,48 @@
 # MessageFormat 2.0 Default Function Registry
 
-This section describes the functions for which each implementation MUST provide
-a _function handler_ to be conformant with this specification.
-
-Implementations MAY implement additional _functions_ or additional _options_.
-In particular, implementations are encouraged to provide feedback on proposed
-_options_ and their values.
+This section defines the **standard** _functions_ which are REQUIRED for conformance with this specification,
+along with **optional** _functions_ that SHOULD be implemented to support
+additional functionality.
+
+To **_<dfn>accept</dfn>_** a function means that an implementation MUST NOT
+emit an _Unknown Function_ error for that _function_'s _identifier_.
+To _accept_ an option means that an implementation MUST NOT
+emit a _Bad Option_ error for that _option_'s _identifier_ when used with the _function_
+it is defined for
+and MUST NOT emit a _Bad Option_ error for any of the _option_ values
+(both **standard** and **optional**)
+defined for that _option_.
+Accepting a _function_ or its _options_ does not mean that a particular output is produced.
+
+_Functions_ can define _options_. 
+An _option_ can be **standard** or **optional**.
+
+Implementations MUST _accept_ each **standard** _function_ and
+MUST _accept_ all _options_ and _option_ values defined as **standard** for those _functions_.
+
+Implementations SHOULD _accept_ each **optional** _function_. 
+For each such _function_, the implementation MUST accept all _options_ and _option_ values 
+listed as **standard** for that _function_.
+
+Implementations SHOULD _accept_ _options_ and _option_ values that are marked as **optional**.
+
+Implementations MAY _accept_ _functions_ not defined in this specification.
+Functions not defined by any version of this specification MUST use an implementation-defined _namespace_.
+In addition, implementations SHOULD provide mechanisms for users to
+register and use user-defined _functions_ and their associated _functional handlers_.
```

Comment:
Would this not mean that any user-defined function "MUST use an implementation-defined _namespace_"? That seems rather arbitrary.

I do not think that we ought to place any such restrictions on user-defined functions.

Reply:
So... SHOULD use an implementation defined namespace, right?

Reply:
That's weird too; the namespace used by a user-defined function should be entirely user-definable, and we should continue to allow for user functions using no namespace at all.

Reply:
> Functions not defined by any version of this specification SHOULD use an implementation-defined or user-defined _namespace_.

Does that fix it? (I added _or user-defined_)

I agree with you that impls/users can do whatever they want, but we should provide guidance not to use un-namespaced functions that we don't define.

Reply:
(mutated the text: have a look)